src/RunspaceProviders/Kubernetes/VMware.ScriptRuntimeService.K8sRunspaceProvider/K8sRunspaceProvider.cs

```diff
@@ -497,122 +507,267 @@ public class K8sRunspaceProvider : IRunspaceProvider {
+            e.Reason.Equals("RELOAD", StringComparison.InvariantCultureIgnoreCase) &&
+            e.Message.Equals("NGINX reload triggered due to a change in configuration");
+      }
+
+      private V1Pod WaitForPodCreation(string name) {
+         V1Pod pod = null;
+         try {
+            _logger.LogDebug($"Waiting k8s Pod '{name}' to become ready");
+            _logger.LogDebug($"K8s API Call ReadNamespacedPod: {name}");
+            pod = _client.CoreV1.ReadNamespacedPod(name, _namespace);
+         } catch (Exception exc) {
+            LogException(exc);
+            throw new RunspaceProviderException(
+                  string.Format(
+                     Resources.K8sRunspaceProvider_WaitCreateComplation_PodNotFound, name),
+                  exc);
+         }
+
+         if (pod == null) {
+            throw new RunspaceProviderException(
+                  string.Format(
+                     Resources.K8sRunspaceProvider_WaitCreateComplation_PodNotFound, name));
+         }
+
+         // Set 10 minutes timeout for container creation.
+         // Worst case would be image pulling from server.
+         int maxRetryCount = 6000;
+         int retryIntervalMs = 100;
+         int retryCount = 1;
+
+         // Wait Pod to become running and obtain IP Address
+         _logger.LogDebug($"Start wating K8s Pod to become running: {pod.Metadata.Name}");
+         // There are three possible phases of a POD
+         // Pending - awaiting containers to start
+         // Running - Pod is initialized and all containers in the Pod are running or completed successfully
+         // Terminating - Pod is terminating
+
+         // The Pending phase could last forever when container image pull error occurred or some other error
+         // in the container initialization happens. In order to stop waiting below we first monitor for Pod status to
+         // phase to switch from pending to running. While Pod is pending phase we monitor the container
+         // creation for errors and if such occur we break the waiting with error.
+         //
+         // The Container creation errors that are related to image pulling failura are stored in the
+         // pod.Status.ContainerStatuses[0].State.Waiting propery is not null which is instance of V1ContainerStateWaiting
+         // The errors are returned as strings in Reason property of the V1ContainerStateWaiting
+         // The strings that represent errors are:
+         //
+         // ImagePullBackOff - Container image pull failed, kubelet is backing off image pull
+         // ImageInspectError - Unable to inspect image
+         // ErrImagePull - General image pull error
+         // ErrImageNeverPull - Required Image is absent on host and PullPolicy is NeverPullImage
+         // RegistryUnavailable - Get http error when pulling image from registry
+         // InvalidImageName - Unable to parse the image name.
+
+         while (
+            pod != null &&
```

Comment:
Put brackets to indicate the order of the operators && and ||

Reply:
fixed